File: Examples/Features/Shape/ConfigShapeWithPictureFill.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Reflection;

namespace GrapeCity.Documents.Excel.Examples.Features.Shape
{
    public class ConfigShapeWithPictureFill : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            IWorksheet worksheet = workbook.Worksheets[0];
            GrapeCity.Documents.Excel.Drawing.IShape shape = worksheet.Shapes.AddShape(GrapeCity.Documents.Excel.Drawing.AutoShapeType.Parallelogram, 20, 20, 200, 100);

            System.IO.Stream stream = this.GetResourceStream("logo.png");

            shape.Fill.UserPicture(stream, GrapeCity.Documents.Excel.Drawing.ImageType.JPG);
            shape.PictureFormat.ColorType = GrapeCity.Documents.Excel.Drawing.PictureColorType.Grayscale;
            shape.PictureFormat.Brightness = 0.6;
            shape.PictureFormat.Contrast = 0.3;
            shape.PictureFormat.Crop.PictureOffsetX = 10;
            shape.PictureFormat.Crop.PictureOffsetY = -5;
            shape.PictureFormat.Crop.PictureWidth = 120;
            shape.PictureFormat.Crop.PictureHeight = 80;
        }

        public override bool ShowViewer
        {
            get
            {
                return false;
            }
        }

        public override bool ShowScreenshot
        {
            get
            {
                return true;
            }
        }

        public override string[] UsedResources
        {
            get
            {
                return new string[] { "logo.png" };
            }
        }
    }
}

File: Examples.VB/Features/Shape/ConfigShapeWithPictureFill.vb
Namespace Features.Shape
    Public Class ConfigShapeWithPictureFill
        Inherits ExampleBase
        Public Overrides Sub Execute(workbook As Excel.Workbook)
            Dim worksheet As IWorksheet = workbook.Worksheets(0)

            Dim shape As IShape = worksheet.Shapes.AddShape(AutoShapeType.Parallelogram, 20, 20, 200, 100)
            Dim stream As IO.Stream = GetResourceStream("logo.png")

            shape.Fill.UserPicture(stream, ImageType.JPG)
            shape.PictureFormat.ColorType = PictureColorType.Grayscale
            shape.PictureFormat.Brightness = 0.6
            shape.PictureFormat.Contrast = 0.3
            shape.PictureFormat.Crop.PictureOffsetX = 10
            shape.PictureFormat.Crop.PictureOffsetY = -5
            shape.PictureFormat.Crop.PictureWidth = 120
            shape.PictureFormat.Crop.PictureHeight = 80
        End Sub
        Public Overrides ReadOnly Property ShowViewer As Boolean
            Get
                Return False
            End Get
        End Property
        Public Overrides ReadOnly Property ShowScreenshot As Boolean
            Get
                Return True
            End Get
        End Property
    End Class
End Namespace
